src/octonode/client.coffee

```diff
@@ -142,37 +144,106 @@ class Client
         body = JSON.parse(body || '{}')
       catch err
         return callback(err)
-
+    debugger
     return callback(new HttpError(body.message, res.statusCode, res.headers, body)) if body.message and res.statusCode in [400, 401, 403, 404, 410, 422]
     callback null, res.statusCode, body, res.headers
 
+  errorHandlePromise: (response, cb) ->
+    debugger
+    {statusCode, body, headers} = response
+
+    fiveError = new HttpError('Error ' + statusCode, statusCode, headers)
+    fourError = new HttpError(body.message, statusCode, headers, body)
+
+    return cb(fiveError) if Math.floor(statusCode/100) is 5
+    return cb(fourError) if body.message and Math.floor(statusCode/100) is 4
+
+    return cb(null, statusCode, body, headers)
+
   # Github api GET request
-  get: (path, params..., callback) ->
-    @request @requestOptions(
+
+  get: (path, params..., cb) =>
+
+    cb = if cb then cb else () -> arguments
```

Comment:
Actually, if you want this to work for promises, it would have to reject the promise if there is an error and it would have to resolve/return an array of all the results if it is successful...  I'll let you take a stab at figuring out how to do that